src/tools/immutable_object_cache/ObjectCacheStore.cc

```diff
@@ -16,6 +16,25 @@ namespace efs = std::experimental::filesystem;
 namespace ceph {
 namespace immutable_obj_cache {
 
+namespace {
+
+class SafeTimerSingleton : public SafeTimer {
```

Comment:
Nit: this is using an ASIO engine so it would be better to use an ASIO timer.

Reply:
In fact,  I don't quite understand why we need to define a singleton timer here, and we only use the lock to new TokenBucketThrottle. About ASIO timer I only know boost:: ASIO:: deadline_ Timer, I am sorry that I don't know how to do here.

Reply:
I think the `TokenBucketThrottle` could be tweaked to no longer accept a `SafeTimer` and `ceph::mutex` as a parameter but instead take a `std::function` for scheduling a callback (a function that takes a lambda and returns an `Event`-like descriptor to allow the timer to be canceled). However, perhaps that's a task for a future PR.